src/Security/Voter/HandleCommandVoter.php

```diff
@@ -64,13 +64,15 @@ protected function supports($attribute, $subject)
     protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
     {
         $allowedRoles = $this->getAllowedRoles(get_class($subject));
+        $actualRoles = $this->roleHierarchy->getReachableRoles($token->getRoles());
 
-        if (count($allowedRoles) > 0) {
-            return $this->decisionManager->decide($token, $allowedRoles);
-        } else {
-            // default conclusion is access denied
-            return false;
+        foreach ($actualRoles as $role) {
+            if (in_array($role->getRole(), $allowedRoles)) {
+                return true;
+            }
         }
```

Comment:
IMO using the decision manager is the right way to check for roles because it allow people to override the way roles are checked in an application either by changing the decision strategy or by completely override the role voter.